incubator/hnc/pkg/forest/forest.go

```diff
@@ -233,11 +233,25 @@ func (ns *Namespace) HasCondition() bool {
 	return len(ns.conditions) > 0
 }
 
+// HasCritCondition returns if the namespace has any critical condition.
+func (ns *Namespace) HasCritCondition() bool {
+	return ns.GetCondition(Local) != nil
+}
+
 // ClearConditions clears local conditions in the namespace.
 func (ns *Namespace) ClearConditions(key string) {
 	delete(ns.conditions, key)
 }
 
+// GetCondition gets a condition list from a key string. It returns nil, if the key doesn't exist.
+func (ns *Namespace) GetCondition(key string) []condition {
+	conditions, ok := ns.conditions[key]
+	if !ok {
```

Comment:
Do we need this? If `ok==false`, won't `conditions` be nil anyway? So you could say:

```
c, _ := ns.conditions[key]
return c
```

Note that very short variable names are actually preferred (https://github.com/golang/go/wiki/CodeReviewComments#variable-names).

Reply:
Good to know! Thanks